lessons/beginners/str/index.md

```diff
@@ -1,353 +1,317 @@
+```pycon
+>>> print("Vtom vnuk křik': \"Hleď!\"")
+Vtom vnuk křik': "Hleď!"
+>>> print('"Jen ho nech," řek\' děd. "Kdo zná líp kraj?"')
+"Jen ho nech," řek' děd. "Kdo zná líp kraj?"
 ```
 
-Záporná čísla vybírají písmenka od konce.
+Ve výsledném řetězci pak ovšem žádné zpětné lomítko *není*.
+Sekvence `\'` je jen způsob, jak v Pythonu zadat `'` – jediný znak.
+Tomu je celkem důležité porozumět.
+Zkus si, jestli zvládneš předpovědět výsledek těchto výrazů:
 
-```plain
-   [0] [1] [2] [3] [4] [5] [6] [7]
-   [-8][-7][-6][-5][-4][-3][-2][-1]
-  ╭───┬───┬───┬───┬───┬───┬───┬───╮
-  │ Č │ o │ k │ o │ l │ á │ d │ a │
-  ╰───┴───┴───┴───┴───┴───┴───┴───╯
+```pycon
+>>> print(".\".")
+>>> len(".\".")
+>>> ".\"."
 ```
-{% endfilter %}
-
-Řetězce umí i jiné triky.
-Třeba můžeš zjistit, jak je řetězec dlouhý
-nebo jestli v sobě obsahuje daný menší řetězec.
-
-<table class="table">
-    <tr>
-        <th>Zápis</th>
-        <th>Popis</th>
-        <th>Příklad</th>
-    </tr>
-    <tr>
-        <td><code>len(r)</code></td>
-        <td>Délka řetězce</td>
-        <td><code>len('čokoláda')</code></td>
-    </tr>
-    <tr>
-        <td><code>x&nbsp;in&nbsp;r</code></td>
-        <td>True pokud je řetězec <code>x</code> obsažen v <code>r</code></td>
-        <td><code>'oko' in 'čokoláda'</code></td>
-    </tr>
-    <tr>
-        <td><code>x&nbsp;not&nbsp;in&nbsp;r</code></td>
-        <td>Opak <code>x in r</code></td>
-        <td><code>'dub' not in 'čokoláda</code></td>
-    </tr>
-</table>
-
-Řetězce vždy berou v potaz velikost písmen,
-takže např. `'ČOKO' in 'čokoláda'` je `False`.
-Kdybys chtěl{{a}} porovnávat bez ohledu na velikost písmen,
-musel{{a}} bys oba řetězce převést třeba na malá písmena
-a pak je porovnat.
-
-A jak se převádí na malá písmena?
-K tomu budeme potřebovat další novou vlastnost Pythonu: metody.
-
-## Metody
-
-*Metoda* (angl. *method*) je jako funkce – něco, co se dá zavolat.
-Na rozdíl od funkce je svázaná s nějakým *objektem* (hodnotou).
-Volá se tak, že se za objekt napíše tečka,
-za ní jméno metody a za to celé se, jako u funkcí, připojí závorky
-s případnými argumenty.
-
-Řetězcové metody `upper()` a `lower()`
-převádí text na velká, respektive malá písmena.
 
-```python
-retezec = 'Ahoj'
-print(retezec.upper())
-print(retezec.lower())
-print(retezec)
+{% filter solution %}
+```pycon
+>>> print(".\".")
+.".
+>>> len(".\".")
+3
+>>> ".\"."
+'.".'
 ```
+{% endfilter %}
 
-> [note]
-> Všimni si, že původní řetězec se nemění; metoda vrátí nový řetězec, ten
-> starý zůstává.
->
-> To je obecná vlastnost řetězců v Pythonu: jednou existující řetězec se už
-> nedá změnit, dá se jen vytvořit nějaký odvozený.
 
+Znaků, které se zadávají sekvencí se zpětným lomítkem je více.
+Jedna ze zajímavějších je `\t`, představující tabulátor – jediný znak, který
+se, když ho vypíšeš, „roztáhne“ na víc mezer.
 
-### Iniciály
+```pycon
+>>> print("a\tb")   # Výpis "pro lidi"
+a       b
+>>> "a\tb"          # Výpis "pro programátory"
```

Comment:
`Výpis "pro programátor(k)y"` ... I suggest fixing of all occurrences (I mean those which are part of this PR)

Reply:
IMO, that would be a suggestion for all the texts on naucse.
I don't see `(k)` used anywhere else yet. Generally I try to address the user with `{{a}}` (hoping that in the future, the page can adjust for logged-in users), but in cases like these use the Czech default.